scripts/gulp-utils.js

```diff
@@ -289,6 +289,45 @@ exports.hoistScssVariables = function() {
   });
 };
 
+/**
+ * Find Sass @use module imports and ensure that they are at the very top of the Sass prior to
+ * running it through the Sass compiler. This also deduplicates @use statements to avoid errors.
+ */
+exports.hoistScssAtUseStatements = function() {
```

Comment:
Maybe I'm misunderstanding which files run through this script. Why not move the `@use` statements in the source code?

Reply:
It's because we have a number of different Sass builds which use different sets of files. PR https://github.com/angular/material/pull/12112 already attempted to just move the many `@use` statements to a single statement in `_variables.scss`, and that worked for the `build-scss` builds, but it broke the module (js and closure) builds.

Reply:
It's also needed because of how the Gulp scripts combine and concatenate the different `.scss` files.

There are probably some optimizations that can be made via re-writing the entire Sass toolchain in this project, but I decided that was out of scope at this time (near end of LTS).